interfaces/builtin/unity8_calendar.go

```diff
@@ -0,0 +1,130 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const unity8CalendarPermanentSlotAppArmor = `
+# Description: Allow operating as the EDS service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+dbus (bind)
+	bus=session
+	name="org.gnome.evolution.dataserver.Calendar7",
+dbus (bind)
+	bus=session
+	name=org.gnome.evolution.dataserver.Subprocess.Backend.*,
+
+########################
+# Calendar
+########################
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarFactory,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarFactory
+	interface=org.gnome.evolution.dataserver.CalendarFactory,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarFactory
+	interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarView/**,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarView/**
+	interface=org.gnome.evolution.dataserver.CalendarView,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/CalendarView/**
+	interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/Backend/Calendar/**
+	interface=org.gnome.evolution.dataserver.Subprocess.Backend,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/Backend/Calendar/**
+	interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/**
+	interface=org.gnome.evolution.dataserver.Calendar,
+dbus (receive, send)
+	bus=session
+	path=/org/gnome/evolution/dataserver/Subprocess/**
+	interface=org.freedesktop.DBus.*,
+`
+
+const unity8CalendarConnectedPlugAppArmor = `
+# Description: Can access the calendar. This policy group is reserved for
+#  vetted applications only in this version of the policy. Once LP: #1227824
+#  is fixed, this can be moved out of reserved status.
+# Usage: reserved
+
+dbus (receive, send)
+     bus=session
+     path=/org/gnome/evolution/dataserver/CalendarFactory
+     peer=(label=unconfined),
+dbus (receive, send)
+     bus=session
+     path=/org/gnome/evolution/dataserver/Subprocess/**
+     peer=(label=unconfined),
+dbus (receive, send)
+     bus=session
+     path=/org/gnome/evolution/dataserver/CalendarView/**
+     peer=(label=unconfined),
+
+# LP: #1319546. Apps shouldn't talk directly to sync-monitor, but allow it for
+# now for trusted apps until sync-monitor is integrated with push
+# notifications.
+dbus (receive, send)
+     bus=session
+     path=/synchronizer{,/**}
+     peer=(label=unconfined),
+`
```

Comment:
These will only work for classic systems where eds is shipped as a deb (ie, `peer=(label=unconfined)` says that the connected client can connected to any unconfined process that uses these paths/interfaces/methods/etc).

Is this interface expected to work on classic systems? If so, then you'll want to move these unconfined rules to unity8CalendarConnectedPlugAppArmor. See the ofono interface for an example.

Please adjust unity8CalendarConnectedPlugAppArmor to use ###SLOT_SECURITY_TAGS### for the peer label (again, the ofono interface can by a guide).